src/Form.js

```diff
@@ -2,20 +2,114 @@ import React from "react";
 import "./Form.css";
 
 class Form extends React.Component {
+  constructor(){
+    super();
+    this.state = {
+      inputs: "",
+      operation: "",
+      calculation: "0"
+    }
+  }
+
+  calcSum = (arr) => {
+    let sum = 0;
+    for(let num of arr){
+      sum = sum + num;
+    }
+    return sum
+  }
+
+  calcMode = (arr) => {
+    let countObj = {};
+    for(let num of arr){
+      if(countObj[num]){
+        countObj[num]++;
+      }else{
+        countObj[num] = 1;
+      }
+    }
+    let keys = Object.keys(countObj);
+    let highestValue = 0;
+    let highestKey;
+    for(let key of keys){
+      if(countObj[key] > highestValue){
+        highestValue = countObj[key];
+        highestKey = key;
+      }
+    }
+    return highestKey
+  }
+
+  handleSubmit = (event) => {
+    event.preventDefault();
+    let numStrArr = ['0','1','2','3','4','5','6','7','8','9'];
+    if(this.state.inputs === "" || !numStrArr.includes(this.state.inputs[0])){
+      this.setState({
+        calculation: 'Invalid input.'
+      })
```

Comment:
```suggestion
   let numArr = this.state.inputs.split(",");  //turn input into array of string nums
    let allValid = numArr.every((num) => parseInt(num) && Number(num));  //check if every element is a valid number
    if (!allValid) {
      this.setState({
        calculation: "Invalid input.",
      });
```
The way you're checking for valid inputs isn't foolproof. If I entered "n2, 2" into your calculator, I would see the error message "Invalid input.", which is great! But if I enter "2n, 2", there is no error message and instead the result is set to "NaN", which should instead be intercepted with an error message. 

To make the validation reliable, you can convert your user's input into an array using `.split(',')`. By splitting on each comma, you'll end up with an array of **just** number strings (no commas).

Then you can use `.every` to make sure that **every single element** is a valid number by checking each one with `parseInt` AND `Number()`, because both of them have loopholes for non-numbers to get through, but if you use **both together** it should cover all your bases. Also keep in mind that this works because "NaN" is a falsy value. If even one element is not a valid number (meaning someone inputed letters or nothing but commas), then an error message "Invalid input.", will be displayed.

Reply:
Thank you for this, I was thinking about how I could test each value but got a bit stuck on the commas. Always forget about .split(',')!